src/metaswitch/clearwater/config_manager/config_access.py

```diff
@@ -422,85 +460,59 @@ def delete_outdated_config_files():
     """Verifies the config, then uploads it to etcd."""
-    validate_config(local_store, config_type, force)
+    validate_config(selected_config, force)
 
     # An exception should have been thrown if validation fails, so we should
     # only reach this point if the config has been validated successfully.
-    upload_config(autoconfirm, config_loader, config_type, force, local_store)
+    upload_config(autoconfirm, config_loader, selected_config.name,
+                  force, local_store)
 
 
-def validate_config(local_store, config_type, force=False):
+def validate_config(selected_config, force=False):
     """
     Validates the config by calling all scripts in the validation folder.
     """
-    script_dir = os.listdir(VALIDATION_SCRIPTS_FOLDER)
-
-    # We can only execute scripts that have execute permissions.
-    scripts_to_run = []
-    for script in script_dir:
-        if os.access(os.path.join(VALIDATION_SCRIPTS_FOLDER, script), os.X_OK):
-            scripts_to_run.append(os.path.join(VALIDATION_SCRIPTS_FOLDER,
-                                               script))
-        else:
-            # Print a warning for each script that isn't being run because of
-            # execute permissions not being set.
-            log.warning("Skipping script %s", script)
-            print ("Validation script {} will be skipped because it does not "
-                   "have execute permissions")
-
-    failed_scripts = []
-    for script in scripts_to_run:
-        try:
-            log.debug("Running validation script %s", script)
-            # Pass in the location of the configuration to check as a parameter
-            # as some scripts need the user to specify which config to
-            # validate.
-            subprocess.check_output([script,
-                                     local_store.config_location(config_type)])
-
-        except subprocess.CalledProcessError as exc:
-            log.error("Validation script %s failed with output:\n %s",
-                      os.path.basename(script),
-                      exc.output)
-
-            # We want to run through all the validation scripts so we can tell
-            # the user all of the problems with their config changes, so don't
-            # bail out of the loop at this point, just record which scripts
-            # have failed.
-            failed_scripts.append(script)
+
+    # This next bit calls to the .validate of the ConfigType, with the
+    # scripts to be used for validation varying depending on the config type.
+    failed_scripts = selected_config.validate()
 
     # When we write the bash injection script, it should either be invoked here
     # or be placed in the VALIDATION_SCRIPTS_FOLDER directory to be executed
-    # in the above loop.
+    # in the above loop. The VALIDATION_SCRIPTS_FOLDER is in the ConfigType
+    # base class so see there, this is only ran through for shared_config
+    # validation
```

Comment:
Rather than trying to fix this comment, I'd just get rid of it entirely. It's not normally good practice to leave comments about the past/future in code, because they're liable to become out of date very easily!

When someone does come to add something to prevent bash injection, they'll be able to look through our lovely, easy-to-read code and reach their own conclusions about where to make their changes. ;)